File: 696-count-binary-substrings/count-binary-substrings.java
class Solution {
    public int countBinarySubstrings(String s) {
     Integer n=s.length();
     ArrayList<Integer>ls=new ArrayList<>();
     int i=0;
     while(i<n)
     {
        int j=i;
        while(j<n && s.charAt(j)==s.charAt(i))
        {
            j++;
        }
        ls.add(j-i);
        i=j;
     } 
     Integer ans=0;
     for(int k=0;k<ls.size()-1;k++)
     {
      ans+=Math.min(ls.get(k),ls.get(k+1));
     }  
     return ans;
    }
}

File: 696-count-binary-substrings/count-binary-substrings.cpp
class Solution {
public:
    int countBinarySubstrings(string s) {
     vector<int>dp;
     int i=0;
     while(i<s.size())
     {
        int j=i;
        while(j<s.size() && s[i]==s[j])
        {
            j++;
        }
        dp.push_back(j-i);
        i=j;
     } 
     int ans=0;
     for(int i=0;i<dp.size()-1;i++)
     {
        ans+=min(dp[i],dp[i+1]);
     }  
     return ans;
    }
};